generic3g/FieldUtilities.F90

```diff
@@ -0,0 +1,131 @@
+#include "MAPL_Generic.h"
+
+module mapl3g_FieldUtilities
+   use MAPL_FieldPointerUtilities, only: FieldGetLocalElementCount
+   use mapl_ErrorHandling
+
+   use esmf
+   implicit none
+   private
+
+   public :: MAPL_FieldReallocate
+
+   interface MAPL_FieldReallocate
+      procedure :: reallocate
+   end interface MAPL_FieldReallocate
+
+   interface operator(==)
+      procedure :: ESMF_GeomEqual
+   end interface operator(==)
+
+   interface operator(/=)
+      procedure :: ESMF_GeomNotEqual
+   end interface operator(/=)
+
+contains
+
+
+   subroutine reallocate(field, geom, typekind, ungriddedUBound, rc)
+      type(ESMF_Field), intent(inout) :: field
+      type(ESMF_Geom), optional, intent(in) :: geom
+      type(ESMF_TypeKind_Flag), optional, intent(in) :: typekind
+      integer, optional, intent(in) :: ungriddedUBound(:)
+      integer, optional, intent(out) :: rc
+
+      integer :: status
+      type(ESMF_TypeKind_Flag) :: old_typekind, typekind_
+      type(ESMF_Geom) :: old_geom, geom_
+      logical :: skip_reallocate
+      integer :: ungriddedDimCount, rank
+      integer, allocatable :: localElementCount(:)
+      integer, allocatable :: old_ungriddedUBound(:)
+      integer, allocatable :: ungriddedUBound_(:), ungriddedLBound_(:)
+      integer :: i
+
+      skip_reallocate = .true.
+
+      call ESMF_FieldGet(field, typekind=old_typekind, geom=old_geom, ungriddedDimCount=ungriddedDimCount, rank=rank, _RC)
+      localElementCount = FieldGetLocalElementCount(field, _RC)
+      old_ungriddedUBound = localElementCount(rank-ungriddedDimCount+1:)
```

Comment:
What value do you _**intend**_ to calculate here?

Reply:
the UBound argument that was used in the original FieldEmptyComplete() call so that I can reuse in the new FieldEmptyComplete().  The ESMF calls are unfortunately not symmetric.  The complete() is just the ungridded, but the FieldGet() is all dims.

Reply:
OK.